test/functional/transactions/0.tx.js

```diff
@@ -0,0 +1,1090 @@
+					badTransactions.push(transaction);
+					done();
+				}, true);
+			});
+
+			it('using array should fail', function (done) {
+				transaction.signature = [];
+
+				sendTransaction(transaction, function (err, res) {
+					node.expect(res).to.have.property('success').to.not.be.ok;
+					node.expect(res).to.have.property('message').to.equal('Invalid transaction body - Failed to validate transaction schema: Expected type string but found type array');
+					badTransactions.push(transaction);
+					done();
+				}, true);
+			});
+
+			it('using object should fail', function (done) {
+				transaction.signature = {};
+
+				sendTransaction(transaction, function (err, res) {
+					node.expect(res).to.have.property('success').to.not.be.ok;
+					node.expect(res).to.have.property('message').to.equal('Invalid transaction body - Failed to validate transaction schema: Expected type string but found type object');
+					badTransactions.push(transaction);
+					done();
+				}, true);
+			});
+
+			it('using string with bad format should fail', function (done) {
+				transaction.signature = 'wrong signature';
+
+				sendTransaction(transaction, function (err, res) {
+					node.expect(res).to.have.property('success').to.not.be.ok;
+					node.expect(res).to.have.property('message').to.equal('Invalid transaction body - Failed to validate transaction schema: Object didn\'t pass validation for format signature: '+transaction.signature);
+					badTransactions.push(transaction);
+					done();
+				}, true);
+			});
+		});
+
+		describe('sending transactions with wrong ID values', function () {
+
+			it('using null should be OK', function (done) {
+				transaction.id = null;
+
+				sendTransaction(transaction, function (err, res) {
+					node.expect(res).to.have.property('success').to.be.ok;
+					node.expect(res).to.have.property('transactionId').to.not.null;
+					transaction.id = res.transactionId;
+					goodTransactions.push(transaction);
+					done();
+				}, true);
+			});
+
+			it('deleting the key should fail', function (done) {
+				delete transaction.id;
+
+				sendTransaction(transaction, function (err, res) {
+					node.expect(res).to.have.property('success').to.be.ok;
+					node.expect(res).to.have.property('transactionId').to.not.null;
+					transaction.id = res.transactionId;
+					goodTransactions.push(transaction);
+					done();
+				}, true);
+			});
+
+			it('using undefined should be OK', function (done) {
+				transaction.id = undefined;
+
+				sendTransaction(transaction, function (err, res) {
+					node.expect(res).to.have.property('success').to.be.ok;
+					node.expect(res).to.have.property('transactionId').to.not.null;
+					transaction.id = res.transactionId;
+					goodTransactions.push(transaction);
+					done();
+				}, true);
+			});
+
+			it('using NaN should be OK', function (done) {
+				transaction.id = NaN;
```

Comment:
@4miners Core calculates an ID automatically if there is no ID or it is not correct. Is it normal behaviour?